source/sessions/tests/driver-sessions-argument-for-correct-client.yml

```diff
@@ -0,0 +1,253 @@
+        arguments:
+          session: *session1
+          filter: { x: 3 }
+          update: { $set: { x: 2 } }
+        expectError:
+          isClientError: true
+      - name: countDocuments
+        object: *collection0
+        arguments:
+          session: *session1
+          filter: {}
+        expectError:
+          isClientError: true
+      - name: createFindCursor
+        object: *collection0
+        arguments:
+          session: *session1
+          filter: {}
+        expectError:
+          isClientError: true
+      - name: createIndex
+        object: *collection0
+        arguments:
+          session: *session1
+          name: foo
+          keys: { x: 1 }
+        expectError:
+          isClientError: true
+      - name: createSearchIndex
+        object: *collection0
+        arguments:
+          session: *session1
+          model: {}
+        expectError:
+          isClientError: true
+      - name: createSearchIndexes
+        object: *collection0
+        arguments:
+          session: *session1
+          models: []
+        expectError:
+          isClientError: true
+      - name: dropIndex
+        object: *collection0
+        arguments:
+          session: *session1
+          name: x_1
+        expectError:
+          isClientError: true
+      - name: dropIndexes
+        object: *collection0
+        arguments:
+          session: *session1
+        expectError:
+          isClientError: true
+      - name: dropSearchIndex
+        object: *collection0
+        arguments:
+          session: *session1
+          name: x_1
+        expectError:
+          isClientError: true
+      - name: listIndexes
+        object: *collection0
+        arguments:
+          session: *session1
+        expectError:
+          isClientError: true
+      - name: listSearchIndexes
+        object: *collection0
+        arguments:
+          session: *session1
+        expectError:
+          isClientError: true
+      - name: listCollections
+        object: *database0
+        arguments:
+          session: *session1
+        expectError:
+          isClientError: true
+      - name: listCollectionNames
+        object: *database0
+        arguments:
+          session: *session1
+        expectError:
+          isClientError: true
+      - name: dropCollection
+        object: *database0
+        arguments:
+          collection: collection0
+          session: *session1
+        expectError:
+          isClientError: true
+      - name: listDatabases
+        object: *client0
+        arguments:
+          session: *session1
+        expectError:
+          isClientError: true
+  - description: Session argument is for the right client (clientBulkWrite)
+    runOnRequirements:
+      - minServerVersion: "8.0"
+    operations:
+      - name: clientBulkWrite
+        object: *client0
+        arguments:
+          models: []
+          session: *session1
+        expectError:
+          isClientError: true
```

Comment:
Where from we get the list of operations? Should we also test the following commands (I've reviewed C# Driver public API that accepts session as a parameter)?
- MongoClient.DropDatabase
- MongoClient.ListDatabaseNames
- MongoClient.Watch
- MongoDatabase.CreateCollection
- MongoDatabase.CreateView
- MongoDatabase.RenameCollection
- MongoDatabase.RunCommand
- MongoDatabase.Watch
- Collection.AggregateToCollection
- Collection.Watch